lib/linguist/heuristics.yml

```diff
@@ -166,6 +166,8 @@ disambiguations:
     pattern: '(?i:^\s*(\<\?xml|xmlns))'
   - language: Graph Modeling Language
     pattern: '(?i:^\s*(graph|node)\s+\[$)'
+  - language: Gerber Image
+    pattern: '\*\%$'
```

Comment:
So, the only exception in Game Maker Language would be that if you had a string that ends with `*%` somewhere?
E.g.;

```
//my code
my_function('abc'); //weird chars *%
```

This would then be detected as Gerber Image?
I think its quite the edge case though, can't think of this pattern occuring often in GML files. 

Is there a way to check what this change would do to existing GML files? Don't think it will, but still :+)

Reply:
The only way the `*%` would be valid in GameMakerLanguage, would be in a string (which is always single-quoted in GML) or in a comment (which is `//` or `/* */` format).

Maybe if `/*` or `//` occurs, we can safely say its Game Maker Language? Not sure if that would be occuring in the Gerber Image format, but if its not, it might be a nice exception to add.

Reply:
I'd say, use an actual Gerber Image command, like:

~~~gerber-image
G04 COMMENT*
X4Y2405D02*
X1264Y2405D01*
~~~

Patterns to match them, taken proudly from my [hand-written Gerber highlighting grammar](https://github.com/Alhadis/language-pcb/blob/master/grammars/gerber.cson):

~~~regexp
^G04.*?\*$
^[XY][-+]?[0-9]+\*$
~~~
Those patterns need to have their "m" switch enabled so ^ and $ match against the start and end of each line, rather than the entire file.

Reply:
> I think its quite the edge case though, can't think of this pattern occuring often in GML files.

I know nothing about GML but it certainly does sound like a small enough edge case for us to not worry about right now.

I'm happy to merge this if others are.

Reply:
Sorry for the late reply, but as a Game Maker Language comment block actually can start with a *, a % might not be that uncommon;

```gml
`/**
*function description
*% of data returned
*/
```

If `/*`, `*/` or `//` is never used in the Gerber format, then I'd say:

`if "*%" matched AND NOT ("/*" OR "*/" OR "//") then language = Gerber`

That might exclude any false positives.

If the Gerber format could contain characters like that it might be better to look at a different solution

Reply:
@RobQuistNL I'd argue that that's still too much of an edge-case to warrant complicating our heuristics that much. Particularly since this is more readable:

~~~GML
/**
 * function description
 * % of data returned
 */
~~~

In any case, the patterns I gave above won't match the sort of comments you've mentioned.